// The Firmament project
// Copyright (c) 2011-2012 Malte Schwarzkopf <malte.schwarzkopf@cl.cam.ac.uk>
//
// HeartbeatMessage is a simple keep-alive message that reports on a live
// resource. It may also trigger registration and fault recovery, although
// initial registration is triggered explicitly using RegistrationMessage.

package firmament;

import "base/resource_desc.proto";
import "base/machine_perf_statistics_sample.proto";

message HeartbeatMessage {
  required string uuid = 1;
  optional string location = 2;
  optional uint64 capacity = 3;
  optional MachinePerfStatisticsSample load = 4;
  optional ResourceDescriptor res_desc = 5;
}
